plumbing/ntwk/ntwk.go

```diff
@@ -12,14 +13,16 @@ type Network struct {
 	host host.Host
 	*pubsub.Subscriber
 	*pubsub.Publisher
+	routing.IpfsRouting
```

Comment:
This is probably where I feel the least sure.  It seems weird to embed an interface here instead of a struct as the other embeddings are structs.  However embedding an interface nicely allows plumbing to be constructed the same way whether the node is in online or offline mode.

Reply:
blocking: we have done work in pubsub to abstract libp2p away from the filecoin code in order to make testing and upgrading easy. Please add a wrapper in the same way as the local `pubsub` package referenced immediately above. See https://github.com/filecoin-project/go-filecoin/pull/2092

Reply:
that sounds like a good reason to me